presto-spi/src/main/java/com/facebook/presto/spi/block/SingleRowBlockWriter.java

```diff
@@ -184,11 +206,19 @@ public BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus)
     @Override
     public String toString()
     {
-        return format("RowBlock{SingleRowBlockWriter=%d, positionCount=%d", numFields, getPositionCount());
+        if (!fieldBlockBuilderReturned) {
+            return format("RowBlock{SingleRowBlockWriter=%d, fieldBlockBuilderReturned=false, positionCount=%d}", numFields, getPositionCount());
+        }
+        else {
+            return format("RowBlock{SingleRowBlockWriter=%d, fieldBlockBuilderReturned=true}");
+        }
     }
 
     private void checkFieldIndexToWrite()
     {
+        if (fieldBlockBuilderReturned) {
+            throw new IllegalStateException("field block builder has been returned");
```

Comment:
cannot do sequential write after getFieldBlockBuilder is called

Reply:
Good point! I also revised the exception message in `getFieldBlockBuilder` as

```
        if (currentFieldIndexToWrite != 0) {
            throw new IllegalStateException("field block builder can only be obtained before any sequential write has done");
        }

```